### ISSUE TITLE
Share header with multiple response objects

### ISSUE THREAD

**Issue by huangsam** [swagger-api/swagger-spec#369]
Can there not just be a simple header generically at the response level not specific to each of the status codes? For our case, this kind of header is 99% the same across our codebase.

```yaml
---

responses:
  200:
    description: no error
    headers:
      Content-Type:
        description: application/json
        type: string
```

It would be a pain to copy+paste this to each and every routine that supports HTTP JSON responses.

**Comment by webron** [comment]
Unfortunately, no. The headers in the responses is actually a new feature in 2.0, and we hadn't thought of the option to allow a global definition for it.

**Comment by RobDolinMS** [comment]
@webron Given your response above, should we close this issue? /cc @huangsam

**Comment by webron** [comment]
It's still an option we might consider in future versions.

**Comment by whitlockjc** [comment]
Isn't this handled [in 3.0](https://github.com/OAI/OpenAPI-Specification/blob/OpenAPI.next/versions/3.0.md#components-object)  You define your reusable headers at `#/components/headers` and then use a `$ref` to reuse it.  It won't allow you to have a singular reference to define all common headers in a response but then you could probably use `#/components/responses` for that.

**Comment by webron** [comment]
Yeah, but this is not about reusability (which makes it easier) - it's about declaring it only once for all.

**Comment by darrelmiller** [comment]
What happens when someone declares a Content-Type header for all responses and then adds a 204 response?

**Comment by webron** [comment]
@darrelmiller did we not just talk today about disallowing Content-Type headers? :)

**Comment by darrelmiller** [comment]
@webron Fine: content-length, Transfer-encoding, Range, ...  how obscure do I need to get? :-)

**Comment by michael-o** [comment]
Obviously the example is bad. consider the following model:

```
paths:
  /api/v1/modules:
    get:
      tags:
        - Modules
      summary: Lists all available modules
      operationId: listModules
      responses:
        200:
          $ref: "#/components/responses/modulesOk"
        400:
          $ref: "#/components/responses/400"
        401:
          $ref: "#/components/responses/401"
        403:
          $ref: "#/components/responses/403"
        404:
          $ref: "#/components/responses/404"
        406:
          $ref: "#/components/responses/406"
        408:
          $ref: "#/components/responses/408"
        500:
          $ref: "#/components/responses/500"
```

Codes like 401, 403, 404, 406, 408 will repeat through all paths. If one could add this to a generic section and selectively overwrite it, it'd be great. Having 50 paths and adding that block over and over again ist just pain.

**Comment by guptaashwanee** [comment]
Is this still not available in the schemas? This will be the great feature and I was also looking for this for a long time.

**Comment by msolujic** [comment]
let use this discussion to underline two common issues and possible enhancements regarding response headers:
1. having a way to define common response headers for some API once and override only some of them if needed 
2. we miss the ability to include multiple headers in one $ref, and not being forced to have 8 refs for 8 headers each and every time

is this planned for some future OAS spec?

**Comment by skaluva** [comment]
Im also looking for this feature... It really helps..  Infact i want to define both request and response common headers for all APIs that are being exposed..

**Comment by handrews** [comment]
This has come up in the context of #5314.  It is not at all certain that we will do it, but I am tentatively tagging this with 3.3 as we are at least considering it.
